includes/payment-methods/class-wc-stripe-upe-payment-method-alipay.php

```diff
@@ -0,0 +1,94 @@
+<?php
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * The alipay Payment Method class extending UPE base class
+ */
+class WC_Stripe_UPE_Payment_Method_Alipay extends WC_Stripe_UPE_Payment_Method {
+
+	const STRIPE_ID = 'alipay';
+
+	const LPM_GATEWAY_CLASS = WC_Gateway_Stripe_Alipay::class;
+
+	/**
+	 * Constructor for alipay payment method
+	 */
+	public function __construct() {
+		parent::__construct();
+		$this->stripe_id            = self::STRIPE_ID;
+		$this->title                = __( 'Pay with Alipay', 'woocommerce-gateway-stripe' );
+		$this->is_reusable          = false;
+		$this->supported_currencies = [
+			'EUR',
+			'AUD',
+			'CAD',
+			'CNY',
+			'GBP',
+			'HKD',
+			'JPY',
+			'NZD',
+			'SGD',
+			'USD',
+			'MYR',
+		];
+		$this->label                = __( 'Alipay', 'woocommerce-gateway-stripe' );
+		$this->description          = __(
+			'Alipay is a popular wallet in China, operated by Ant Financial Services Group, a financial services provider affiliated with Alibaba.',
+			'woocommerce-gateway-stripe'
+		);
+	}
+
+	/**
+	 * Returns the currencies this UPE method supports for the Stripe account.
+	 * Documentation: https://stripe.com/docs/payments/alipay#supported-currencies.
+	 *
+	 * @return array
+	 */
+	public function get_supported_currencies() {
```

Comment:
As I moved Alipay inside UPE, I was getting an error on the checkout page when I was using an `US` Stripe account with store currency set as `EURO`. Also the UPE payment methods failed to render on the page because of this error.

![error alipay](https://github.com/woocommerce/woocommerce-gateway-stripe/assets/33387139/1184a74c-6f93-47ff-9796-1df4dda939a6)
 
As per my understanding after reading https://stripe.com/docs/payments/alipay#supported-currencies that says `You can create Alipay payments in the currencies that map to your country` I have created this mapping function that returns the supported currency list based on the Stripe account's country. 

Now on the settings page also when UPE is enabled, if your Stripe account is a US account but WC store currency is Euro, it will show Alipay as unsupported for currency.

To the reviewer: please check if my assumption is correct here.